sdks/python/apache_beam/runners/worker/statesampler.py

```diff
@@ -29,6 +31,22 @@
   FAST_SAMPLER = False
 
 
+class ExecutionStateSamplers(threading.local):
+  """ Per-thread state sampler. """
+  def __init__(self):
+    super(ExecutionStateSamplers, self).__init__()
+    self._current_sampler = None
+
+  def current_sampler(self):
+    return self._current_sampler
+
+  def set_sampler(self, sampler):
+    self._current_sampler = sampler
+
+
+EXECUTION_STATE_SAMPLERS = ExecutionStateSamplers()
```

Comment:
Perhaps the name should indicate that this is per-thread? In general, creating a global singleton class rather than module-level functions is a Javaism.